src/sources/postgresql_metrics.rs

```diff
@@ -0,0 +1,836 @@
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_checkpoints_req_total",
+                counter!(row_get_value::<i64>(&row, "checkpoints_req")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_checkpoint_write_time_seconds_total",
+                counter!(row_get_value::<f64>(&row, "checkpoint_write_time")? / 1000f64),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_checkpoint_sync_time_seconds_total",
+                counter!(row_get_value::<f64>(&row, "checkpoint_sync_time")? / 1000f64),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_checkpoint_total",
+                counter!(row_get_value::<i64>(&row, "buffers_checkpoint")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_clean_total",
+                counter!(row_get_value::<i64>(&row, "buffers_clean")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_maxwritten_clean_total",
+                counter!(row_get_value::<i64>(&row, "maxwritten_clean")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_backend_total",
+                counter!(row_get_value::<i64>(&row, "buffers_backend")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_backend_fsync_total",
+                counter!(row_get_value::<i64>(&row, "buffers_backend_fsync")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_alloc_total",
+                counter!(row_get_value::<i64>(&row, "buffers_alloc")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_stats_reset",
+                gauge!(row_get_value::<DateTime<Utc>>(&row, "stats_reset")?.timestamp()),
+                tags!(self.tags),
+            ),
+        ])
+    }
+
+    fn create_metric(
+        &self,
+        name: &str,
+        value: MetricValue,
+        tags: BTreeMap<String, String>,
+    ) -> Metric {
+        Metric {
+            name: name.into(),
+            namespace: self.namespace.clone(),
+            timestamp: Some(Utc::now()),
+            tags: Some(tags),
+            kind: MetricKind::Absolute,
+            value,
+        }
+    }
+}
+
+fn row_get_value<'a, T: FromSql<'a>>(row: &'a Row, key: &'static str) -> Result<T, CollectError> {
+    row.try_get::<&str, T>(key)
+        .map_err(|source| CollectError::PostgresGetValue { source, key })
+}
+
+fn config_to_endpoint(config: &Config) -> String {
```

Comment:
Just to confirm, this will result in endpoints being stripped of the password, if it was set in the URL?

Reply:
Yes, password will not be added to resulted endpoint